_posts/2017-09-11-DevSwap.markdown

```diff
@@ -0,0 +1,109 @@
+
+Some people worked on modernizing codebases with projects like [server-side hot file swapping][modern] and the introduction of a GraphQL endpoint to our consignments app. Others worked on building features like [setting up Slack notifications][feedback] when we receive customer feedback and adding ElasticSearch indexing to projects. We learned new languages [by building features][languages] and updated [many][docs1] [docs][docs2].
+
+All work aside, the fact that we were having conversations with each other about the details of the technology while gaining a fresh perspective of someone else's experience was valuable.
+
+## Feedback from Engineers
+
+I learned a lot and had fun, but I wondered what everyone else thought. At the end of the second day, I sent out a survey to all 26 engineers and received 14 responses. Here are some of those results:
+
+Did you get a chance to collaborate with someone you don't normally work with?
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQoAQpvGiq2-5EJWz__vS0n4Iw9r3rY_eSEaPqKQzUlPnObnono-RFS4qLSjrmGTqGnbMOoQsA1L9WX/pubchart?oid=942845256&amp;format=interactive"></iframe>
+
+<br>
+Did you gain a better understanding of a piece of Artsy Engineering you don't normally interact with?
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQoAQpvGiq2-5EJWz__vS0n4Iw9r3rY_eSEaPqKQzUlPnObnono-RFS4qLSjrmGTqGnbMOoQsA1L9WX/pubchart?oid=543941925&amp;format=interactive"></iframe>
+
+<br>
+If we were to swap again in 6 months, how long should it last?
+
+<iframe width="600" height="371" seamless frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQoAQpvGiq2-5EJWz__vS0n4Iw9r3rY_eSEaPqKQzUlPnObnono-RFS4qLSjrmGTqGnbMOoQsA1L9WX/pubchart?oid=1071239654&amp;format=interactive"></iframe>
+
+<br>
+<br>
+
+> "Would make the swap slightly longer -- was a short timeframe to get through a project."
+
+> "Onboarding took up most of the time. It was hard to get anything beyond that done."
+
+One clear piece of feedback was the desire to spend more time on an assigned team. Some engineers had trouble onboarding and setting up projects within the two days. If engineers can't spin up a dev environment within a day, what does that mean for usability? These are solvable problems — we could think about dockerizing apps or having better READMEs. During the session, some folks went back to their old teams to help new members with problems. While this was great, it meant less time for their own learning. We could alleviate the feeling of being rushed by providing more time and better initial setup. Also, by communicating that this isn't a race or contest, people may feel less pressured to deliver code.
```

Comment:
> Some engineers had trouble...

"Some engineers had trouble onboarding, setting up projects, _and_ accomplishing meaningful work within the 2 days."

> We could alleviate the feeling of being rushed

I actually liked this pressure... it was productive.

Reply:
Thanks, I'll change the first point. For the second point, I think the feelings were mixed, so I'll try to change the language to better reflect that.

Reply:
@joeyAghion what do you think of:

> One option is to extend the duration of the swap. But by extending it too long, it could remove the hackathon mentality that kept people highly focused. It seems like a combination of tweaking the duration, providing better initial setup, and clearly communicating expectations might help for next time.

instead of:

> We could alleviate the feeling of being rushed by providing more time and better initial setup. Also, by communicating that this isn't a race or contest, people may feel less pressured to deliver code.